thrift/thrift-gen/main.go

```diff
@@ -149,3 +200,67 @@ func cleanGeneratedCode(generated []byte) []byte {
 func contextType() string {
 	return "thrift.Context"
 }
+
+func getTemplateFromFile(filename string, AST *parser.Thrift) (*template.Template, error) {
+	// export useful functions to the template renderer
+	state := NewState(AST)
+	funcs := map[string]interface{}{
+		"contextType": contextType,
+		"lowercase": func(in string) string {
+			return strings.ToLower(in)
+		},
+		"goPublicName": goPublicName,
+		"goType": func(t *parser.Type) string {
+			return state.goType(t)
+		},
+	}
+
+	// read template file
+	templateBytes, err := ioutil.ReadFile(filename)
```

Comment:
I was thinking about passing in the absolute path to the file and do the resolution relative to the `GOPATH` in the `Makefile`. However I'm running into some problems with trying that so I might implement it in here. However I think it is also good that one can pass in an absolute path. So the logic would look something like:

1. Resolve `filename` for relative path to `pwd`
2. If file does not exists Iterate over entries in `GOPATH`
  - Check for existence of file
  - Exit iteration if file exists